app/assets/javascripts/dashboard/views/dash/explorative_annotations_view.js

```diff
@@ -0,0 +1,354 @@
+// @flow
+/* eslint-disable jsx-a11y/href-no-hash */
+
+import React from "react";
+import Request from "libs/request";
+import { Spin, Input, Table } from "antd";
+import type { APIAnnotationType } from "admin/api_flow_types";
+import FormatUtils from "libs/format_utils";
+import Toast from "libs/toast";
+import Utils from "libs/utils";
+import update from "immutability-helper";
+import EditableTextLabel from "oxalis/view/components/editable_text_label";
+
+const { Column } = Table;
+const { Search } = Input;
+
+type Props = {
+  userID: ?string,
+  isAdminView: boolean,
+};
+
+export default class ExplorativeAnnotationsView extends React.PureComponent {
+  props: Props;
+  state: {
+    showArchivedTracings: boolean,
+    archivedTracings: Array<APIAnnotationType>,
+    unarchivedTracings: Array<APIAnnotationType>,
+    alreadyFetchedMetaInfo: {
+      archived: boolean,
+      unarchived: boolean,
+    },
+    searchQuery: string,
+    requestCount: number,
+  } = {
+    showArchivedTracings: false,
+    archivedTracings: [],
+    unarchivedTracings: [],
+    alreadyFetchedMetaInfo: {
+      archived: false,
+      unarchived: false,
+    },
+    searchQuery: "",
+    requestCount: 0,
+  };
+
+  componentDidMount() {
+    this.fetchDataMaybe();
+  }
+
+  isFetchNecessary(): boolean {
+    const accessor = this.state.showArchivedTracings ? "archived" : "unarchived";
+    return !this.state.alreadyFetchedMetaInfo[accessor];
+  }
+
+  enterRequest() {
+    this.setState({ requestCount: this.state.requestCount + 1 });
+  }
+
+  leaveRequest() {
+    this.setState({ requestCount: this.state.requestCount - 1 });
+  }
+
+  async fetchDataMaybe(): Promise<void> {
+    if (!this.isFetchNecessary()) {
+      return;
+    }
+
+    const isFinishedString = this.state.showArchivedTracings.toString();
+    const url = this.props.userID
+      ? `/api/users/${this.props.userID}/annotations?isFinished=${isFinishedString}`
+      : `/api/user/annotations?isFinished=${isFinishedString}`;
+
+    this.enterRequest();
+    const tracings = await Request.receiveJSON(url);
+    this.leaveRequest();
+    if (this.state.showArchivedTracings) {
+      this.setState(
+        update(this.state, {
+          archivedTracings: { $set: tracings },
+          alreadyFetchedMetaInfo: {
+            archived: { $set: true },
+          },
+        }),
+      );
+    } else {
+      this.setState(
+        update(this.state, {
+          unarchivedTracings: { $set: tracings },
+          alreadyFetchedMetaInfo: {
+            unarchived: { $set: true },
+          },
+        }),
+      );
+    }
+  }
+
+  toggleShowArchived = () => {
+    this.setState({ showArchivedTracings: !this.state.showArchivedTracings }, () =>
```

Comment:
`setState` is asynchronous which is why the callback is necessary (`fetchDataMaybe` reads from the state and otherwise would read the untoggled state).